docs: Add JetBrains and VSCode authoring tips (#801)
cerbos/cerbos: docs/modules/policies/pages/authoring_tips.adoc

    end,
})
----

==== JetBrains IDEs

Go into the Preferences of JetBrains IDE of choice and navigate to the `JSON Schema Mappings` menu.

----

> Preferences > Languages & Frameworks > Schemas and DTDs > JSON Schema Mappings

Reviewer: I think you can use the Menu macro here: https://docs.asciidoctor.org/asciidoc/latest/macros/ui-macros/#menu-macro-syntax

----

Add an entry with the following configuration

[subs="attributes"]
----
Name: Cerbos
Schema file or URL: {latest-schema-url}
Schema version: JSON Schema Version 7
File path pattern: cerbos/*
----
image:jetbrains-menu.png[alt="JetBrains JSON Schema Mappings menu",role="center-img"]

> NOTE: In the example above, the schema is applied to all files in the /cerbos directory.

Reviewer: I am probably wrong here, but, Isn't the file path pattern relative in the above example? So presumably it applies to any files contained in a directory named cerbos?

```suggestion
NOTE: In the example above, the schema is applied to all files in the cerbos directory.
```

Author: Yes, you are right. My mistake.

==== Visual Studio Code

> Before starting, make sure you know how to modify Visual Studio Code settings by taking a look at the link:https://code.visualstudio.com/docs/getstarted/settings[VSCode documentation].
> 
> For Visual Studio Code, an extension is required to work with the schema.
> Please install the extension from link:https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml[here]
> 
> After the required extension is ready, please navigate to the Visual Studio Code's `settings.json` and add the following part.

Reviewer: Documentation doesn't usually contain words like "please" :)
 
```suggestion
If you are new to Visual Studio Code, refer to the
 link:https://code.visualstudio.com/docs/getstarted/settings[documentation] for more information about how to change settings.

Install the YAML language server extension from https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml

After the extension is installed, hit kbd:[Ctrl+,] or menu:File[Preferences > Settings] to edit settings. Expand menu:Extensions[YAML], click `Edit in settings.json` under `Yaml: Schemas`. and add the following snippet:

```

[source,json,subs="attributes"]
----
{
  "yaml.schemas": {
    "{latest-schema-url}": "cerbos/*"
  }
}
----

> NOTE: In the example above, the schema is applied to all files in the /cerbos directory.

Reviewer: ```suggestion
NOTE: In the example above, the schema is applied to all files in the cerbos directory.
```
